### DISCUSSION TITLE
PM2 cluster mode in DOcker

### DISCUSSION THREAD

**Discussion by smasilamani-cfins** [vercel/next.js#32128] [Help]
Does anyone know how we can run it in cluster mode inside docker? I tried it in fork mode and it works well but we would like to run the app in cluster mode.

My Docker command looks like this : CMD [ "pm2-runtime" ,"start","pm2.json"]

pm2.json file

```
{
  "apps": [
    {
      "name": "App-Client",
      "cwd": "/usr/app/client",
      "script": "node_modules/next/dist/bin/next",
      "args": "start -p 4000",
      "instances": "max",
      "exec_mode": "cluster",
      "kill_timeout": 5000,
      "env": { "NODE_ENV": "production", "production": true }
    }
  ]
}
```

Error: (As you can see its looking for .next folder inside pm2.json folder)

> 2021-12-03T04:31:51: PM2 log: App [App-Client:0] exited with code [1] via signal [SIGINT]
> 2021-12-03T04:31:51: PM2 log: App [App-Client:0] starting in -cluster mode-
> Error: Could not find a valid build in the '/usr/app/client/pm2.json/.next' directory! Try building your app with 'next build' before starting the server.
>     at Server.readBuildId (/usr/app/dip/client/node_modules/next/next-server/server/next-server.ts:1671:15)
>     at new Server (/usr/app/dip/client/node_modules/next/next-server/server/next-server.ts:179:25)
>     at createServer (/usr/app/dip/client/node_modules/next/server/next.ts:41:10)
>     at start (/usr/app/dip/client/node_modules/next/server/lib/start-server.ts:9:15)
>     at exec (/usr/app/dip/client/node_modules/next/cli/next-start.ts:53:3)
>     at /usr/app/dip/client/node_modules/next/bin/next.ts:107:36
> 2021-12-03T04:31:51: PM2 log: App name:App-Client id:1 disconnected
>

**Comment by conanbatt** [comment]
The trace is mentioning you need to run build before start, which is how it works in a production environment.
